src/_imaging.c

```diff
@@ -1242,11 +1255,61 @@ _histogram(ImagingObject* self, PyObject* args)
         PyList_SetItem(list, i, item);
     }
 
+    /* Destroy the histogram structure */
     ImagingHistogramDelete(h);
 
     return list;
 }
 
+static PyObject*
+_entropy(ImagingObject* self, PyObject* args)
+{
+    ImagingHistogram h;
+    PyObject* entropy;
+    int idx, length;
+    long sum;
+    double fentropy, fsum, p;
+    union hist_extrema extrema;
+    union hist_extrema* ep;
+
+    PyObject* extremap = NULL;
+    ImagingObject* maskp = NULL;
+    if (!PyArg_ParseTuple(args, "|OO!", &extremap, &Imaging_Type, &maskp))
+        return NULL;
+
+    /* Using a local var to avoid allocations. */
+    ep = parse_histogram_extremap(self, extremap, &extrema);
+    h = ImagingGetHistogram(self->image, (maskp) ? maskp->image : NULL, ep);
+
+    if (!h)
+        return NULL;
+
+    /* Calculate the histogram entropy */
+    /* First, sum the histogram data */
+    length = h->bands * 256;
+    sum = 0;
+    for (idx = 0; idx < length; idx++) {
+        sum += h->histogram[idx];
+    }
+
+    /* Next, normalize the histogram data, */
+    /* using the histogram sum value */
+    fsum = (double)sum;
+    fentropy = 0.0;
+    for (idx = 0; idx < length; idx++) {
+        p = (double)h->histogram[idx] / fsum;
+        fentropy += p != 0.0 ? (p * log(p) * M_LOG2E) : 0.0;
```

Comment:
Minor: It wouldn't be better to made this an `if` statement, instead of adding zero under some circumstances?
```suggestion
        if (p != 0.0) {
            fentropy += p * log(p) * M_LOG2E;
        }
```

Reply:
That sounds fine to me – the only question I have is that OK, since I am in no way a code-generation expert: which of the two will produce the most efficient set of operations, _aprés_-compilation?

Reply:
… I actually figured this out using [godbolt](https://godbolt.org) – the `if` statement has less branching and seems to be more generally optimizable by the latest GCC and recent Clangs. Merging!